libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -542,30 +541,36 @@ void AP_TECS::_update_height_demand(void)
         // during approach compensate for height filter lag
         if (_flags.is_doing_auto_land) {
             _hgt_dem += _hgt_dem_tconst * _hgt_rate_dem;
```

Comment:
I think we should also reset __max_sink_scaler_ (maybe __max_climb_scaler_  aswell) during  auto landing, because now if we were in Auto mode with restricted sink rate limit to let's say to 0.5 m/s, and then we change mode to Land, UAV will remember last value and not descend properly during Approach stage.

Reply:
Agree. That would apply to platforms that can descend faster during landing approach due to flap or airbrake deployment. I'll implement that.